tools/linker/CoreTypeMapStep.cs

```diff
@@ -25,19 +26,142 @@ namespace MonoTouch.Tuner {
 	public class CoreTypeMapStep : TypeMapStep {
 		HashSet<TypeDefinition> cached_isnsobject = new HashSet<TypeDefinition> ();
 		Dictionary<TypeDefinition, bool?> isdirectbinding_value = new Dictionary<TypeDefinition, bool?> ();
+		bool dynamic_registration_support_required;
 
 		DerivedLinkContext LinkContext {
 			get {
 				return (DerivedLinkContext) base.Context;
 			}
 		}
 
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (LinkContext.App.Optimizations.RemoveDynamicRegistrar != false)
+				dynamic_registration_support_required |= RequiresDynamicRegistrar (assembly, LinkContext.App.Optimizations.RemoveDynamicRegistrar == true);
+
+			base.ProcessAssembly (assembly);
+		}
+
+		// If certain conditions are met, we can optimize away the code for the dynamic registrar.
+		bool RequiresDynamicRegistrar (AssemblyDefinition assembly, bool warnIfRequired)
+		{
+			// Disable removing the dynamic registrar for XM/Classic to simplify the code a little bit.
```

Comment:
XM classic shouldn't be building anymore so technically not needed, but still nice to note.

Reply:
mmp still supports building Classic apps

Reply:
This is true. duh